Manager/src/main/java/io/raspberrywallet/manager/database/KeyPartEntity.java

```diff
@@ -25,6 +31,22 @@ protected synchronized void clean() {
 		}
 		
 		order=Integer.rotateRight(order, order);
-	
+	}
+
+	/*
+	* Needed to override this so `WalletEntity` can be compared with ease.
+	* Two `KeyPartEntity` with different pointer can be equal now.
+	* */
+	@Override
+	public boolean equals(Object obj) {
+		if(obj instanceof KeyPartEntity) {
+			KeyPartEntity kpe = (KeyPartEntity) obj;
+			return (
+					this.module.equals(kpe.module)
+					&& this.order == kpe.order
+					&& Arrays.equals(this.payload, kpe.payload)
+			);
+		}
+		return super.equals(obj);
```

Comment:
We should use there EqualsAndHashCode annotation from Lombok, instead of hard coding this.

Reply:
I'm not convinced if it generates proper `equal` for `List`s, because their example on two arrays uses `Arrays.deepEqual` which only is implemented for array objects.